### PR TITLE
Intan loader

### PR DESCRIPTION
**Pull request by tdincer**
Based on the "One file per channel type format" section of the official intan manual (page 11): https://intantech.com/files/Intan_RHS2000_data_file_formats.pdf

Task list:
- [x] file path (channel) - output connection needs to be done. If pattern returns more than 1 file, the outputs should be clearly associated with the file name. This is optional. Current version works fine if the pattern is a full file name (e.g "amp-B-000.dat").
- [ ] Ideally `read_qstring` and `read_header` functions should be imported from intan. We need to decide on how to operate the intan. I guess the decision will be made with Alessio.
- [x] Do basic reading
- [x] Add docstrings.

For the completeness of the discussion:
- The official example [notebook](https://github.com/Intan-Technologies/load-rhs-notebook-python/blob/main/LoadIntanRHS_Python.ipynb) does not work for Alex's data.

### CODE REVIEW

**Comment by kabilar** [comment]
Thanks @tdincer.  Drafted this pull request since we wouldn't want to add hardcoded paths to this package.

**Comment by tdincer** [comment]
@kabilar This PR is to a new branch. It's meant to keep track of the changes. Iterations should be done forking this PR. Not by copy pasting the content.

**Comment by tdincer** [comment]
@JaerongA and @kabilar I cleaned the code. Let's iterate from here addressing the issues discussed in slack.

**Comment by tdincer** [comment]
@JaerongA What do we need to return to make this loader useful for array-ephys?

**Comment by JaerongA** [comment]
> @JaerongA What do we need to return to make this loader useful for array-ephys?

The original [intan code ](https://github.com/JaerongA/pyfinch/blob/46c437796f9cce00cfe887d4f6d3e311bebfd7f0/pyfinch/utils/intan/load_intan_rhd_format.py#L15) used to ouput everything (including all the parameters) into a single dictionary format. We might needs parameter values as well. To be completely sure of what's really needed, it'd be worth ingesting some sample data in the `element-array-ephys` pipeline.

**Review by JaerongA — APPROVED** [review]

File: element_interface/intan_loader.py [hunk 1]
```diff
@@ -0,0 +1,271 @@
...
+    header["num_amplifier_channels"] = len(header["amplifier_channels"])
+    header["num_board_adc_channels"] = len(header["board_adc_channels"])
+    header["num_board_dac_channels"] = len(header["board_dac_channels"])
+    header["num_board_dig_in_channels"] = len(header["board_dig_in_channels"])
+    header["num_board_dig_out_channels"] = len(header["board_dig_out_channels"])
+
+    return header
+
+
+def load_rhs(folder: str, file_expr: str):
```
Comment (JaerongA) [inline element_interface/intan_loader.py:227]:
```suggestion
def load_rhs(folder: str):
```

I don't think regex need to be specified here as an argument. This is supposed to fetch whatever that's inside the data folder

**Review by JaerongA — COMMENTED** [review]
@tdincer. Looks like this only reads from amp.dat files. Some sessions have digital input and stimulus files.

File: element_interface/intan_loader.py (hunk 1 above)
Comment (tdincer) [reply element_interface/intan_loader.py:227]: It adds flexibility to not read everything. Say the files are `amp-B-000.dat ... to  amp-B-015.dat`, `amp-D-000.dat ... to  amp-D-015.dat` and `board-DIGITAL-IN-01.dat`. 

To read all channels (A and B) in the working directory:
load_rhs(".", "amp*")

To read only the `B` channels in the working directory:
load_rhs(".", "amp-B*")

Comment (JaerongA) [reply element_interface/intan_loader.py:227]: I see. Then we can keep that and set the default to e.g.,`*.dat` (read everything) and allow a user to specify if needed.

**Comment by kabilar** [comment]
> @kabilar This PR is to a new branch. It's meant to keep track of the changes. Iterations should be done forking this PR. Not by copy pasting the content.

Thanks @tdincer.  Whoops, misread the pull request.  You are right.

**Review by JaerongA — APPROVED** [review]

**Pull request merged**